src/webgpu/compat/api/validation/createBindGroup.spec.ts

```diff
@@ -0,0 +1,176 @@
+        case '2d':
+        case '2d-array':
+          return true;
+        case 'cube':
+          return depthOrArrayLayers === 6;
+        case 'cube-array':
+          return depthOrArrayLayers % 6 === 0;
+        default:
+          return false;
+      }
+      break;
+    case '3d':
+      return viewDimension === '3d';
+  }
+}
+
+function isValidDimensionForDepthOrArrayLayers(
+  dimension: GPUTextureDimension | undefined,
+  depthOrArrayLayers: number
+) {
+  switch (dimension) {
+    case '1d':
+      return depthOrArrayLayers === 1;
+    default:
+      return true;
+  }
+}
+
+function isValidViewDimensionForDepthOrArrayLayers(
+  viewDimension: GPUTextureViewDimension | undefined,
+  depthOrArrayLayers: number
+) {
+  switch (viewDimension) {
+    case 'cube':
+      return depthOrArrayLayers === 6;
+    case 'cube-array':
+      return depthOrArrayLayers % 6 === 0;
+    default:
+      return true;
+  }
+  return viewDimension === 'cube';
+}
+
+function getEffectiveTextureBindingViewDimension(
+  dimension: GPUTextureDimension | undefined,
+  depthOrArrayLayers: number,
+  textureBindingViewDimension: GPUTextureViewDimension | undefined
+) {
+  if (textureBindingViewDimension) {
+    return textureBindingViewDimension;
+  }
+
+  switch (dimension) {
+    case '1d':
+      return '1d';
+    case '2d':
+    case undefined:
+      return depthOrArrayLayers > 1 ? '2d-array' : '2d';
+      break;
+    case '3d':
+      return '3d';
+  }
+}
+
+g.test('dimension_matches_textureBindingViewDimension')
+  .desc(
+    `
+    Tests that, in compat mode, the dimension of a view is compatible with a texture's textureBindingViewDimension
+    when used as a TEXTURE_BINDING.
+    `
+  )
+  .params(u =>
+    u //
+      .combine('dimension', [...kTextureDimensions, undefined])
+      .combine('textureBindingViewDimension', [...kTextureViewDimensions, undefined])
+      .combine('viewDimension', [...kTextureViewDimensions, undefined])
+      .combine('depthOrArrayLayers', [1, 2, 6])
+      .filter(
+        ({ dimension, textureBindingViewDimension, depthOrArrayLayers, viewDimension }) =>
+          textureBindingViewDimension !== 'cube-array' &&
+          isTextureBindingViewDimensionCompatibleWithDimension(
+            dimension,
+            textureBindingViewDimension
+          ) &&
+          isValidViewDimensionForDimension(dimension, depthOrArrayLayers, viewDimension) &&
+          isValidViewDimensionForDepthOrArrayLayers(
+            textureBindingViewDimension,
+            depthOrArrayLayers
+          ) &&
+          isValidDimensionForDepthOrArrayLayers(dimension, depthOrArrayLayers)
+      )
+  )
+  .fn(t => {
+    const { dimension, textureBindingViewDimension, viewDimension, depthOrArrayLayers } = t.params;
+
+    const texture = t.device.createTexture({
+      size: [1, 1, depthOrArrayLayers],
+      format: 'rgba8unorm',
+      usage: GPUTextureUsage.TEXTURE_BINDING,
+      ...(dimension && { dimension }),
```

Comment:
weird, but I'm going to have to file this one in my book of JS tricks.

We usually don't bother making sure unused keys won't be set to `undefined` - webidl `dictionary` guarantees that `{}` and `{x: undefined}` are the same. IMO probably not necessary here, but OK.